### PR TITLE
Full Speed Ahead: Various Improvements

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

Tested a bunch, I'm getting pretty good at the minigame...

**New:**
Added tracking via Quest Log
Fixed missing Item and NPC IDs in completion CS
Changed motivation/pep relationship
Added clamping to motivation and pep to fix any weirdness with the bar rendering
Added failure CS when you dismount or run out of motivation, which lets you enable easy mode (or not)
Added easy mode: tracked through the effect's power (acts as a modifier for motivation decay and pep growth)
Fixed bug where the final Raptor Food region was acting like it was Syrillia's region
Added MEET_SYRILLIA text when you've eaten 5 foods

**Not doing in this PR:**
Additional Raptor animations
Varying pep and speed depending on terrain slope etc.
Best time tracking

**Once this PR goes in, I think it would be fair to open two new issues:**
**- [Good First Issue] Quest: Full Speed Ahead! doesn't track server best time**
Using `Get/SetServerVariable`

**- [Good First Issue][Core] Quest: Full Speed Ahead! raptor doesn't slow down as it runs out of pep**
Mount speed is hard-coded, and needs to be modifiable (through mods or otherwise) for Full Speed Ahead!

`battleentity.cpp`
```
uint8 CBattleEntity::GetSpeed()
{
    return (isMounted() ? 50 + map_config.speed_mod : std::clamp<uint16>(speed * (100 + getMod(Mod::MOVE)) / 100, std::numeric_limits<uint8>::min(), std::numeric_limits<uint8>::max()));
}
```

### CODE REVIEW

File: scripts/globals/quests.lua [hunk 1]
```diff
@@ -481,6 +481,8 @@ tpz.quest.id =
         VW_OP_115_VALKURM_DUSTER        = 168,
         VW_OP_118_BUBURIMU_SQUALL       = 169,
         PRELUDE_TO_PUISSANCE            = 170,
+
+        FULL_SPEED_AHEAD                = 179, -- + --
```
Comment (ibm2431) [inline scripts/globals/quests.lua:485]: This gap worries me; we'll have to check dats later and add whatever is missing from Jeuno's quest log

File: scripts/quests/full_speed_ahead.lua [hunk 2]
```diff
@@ -13,25 +30,30 @@ tpz.full_speed_ahead.motivation_decay = 2
...
 
-tpz.full_speed_ahead.onEffectLose = function(player)
+tpz.full_speed_ahead.onEffectLose = function(player, effect)
     player:delStatusEffectSilent(tpz.effect.MOUNTED)
     player:countdown(0) 
     player:enableEntities({})
+
+    -- If in Batallia Downs and didn't get the completion flag (failed/dismounted)
+    if player:getZoneID() == tpz.zone.BATALLIA_DOWNS and player:getCharVar("[QUEST]FullSpeedAhead") ~= 4 then
+        player:startEvent(26, 0, effect:getPower())
```
Comment (ibm2431) [inline scripts/quests/full_speed_ahead.lua:50]: How are we handling the quest mounted status again? Did it get a special effect, or are we using regular mounted?

In other words: what if a player fails, quits, rides a chocobo in Batalia, then dismounts?

File: scripts/zones/Upper_Jeuno/npcs/Mapitoto.lua [hunk 3]
```diff
@@ -31,15 +32,15 @@ function onTrade(player,npc,trade)
...
-    local hasTrainersWhistle = player:hasKeyItem(tpz.ki.TRAINERS_WHISTLE)
 
-    if hasTrainersWhistle then
+    if fsaQuest == QUEST_COMPLETED then
         player:startEvent(10226)
-    elseif fullSpeedAheadStatus == 2 then -- Retry
-        player:startEvent(10224, 1)
     elseif fullSpeedAheadStatus == 3 then -- Complete
-        player:startEvent(10225, tpz.ki.TRAINERS_WHISTLE)
+        player:startEvent(10225, tpz.ki.TRAINERS_WHISTLE, 15533, ID.npc.MAPITOTO)
```
Comment (ibm2431) [inline scripts/zones/Upper_Jeuno/npcs/Mapitoto.lua]: Could be wrong, but I think you bumped up your quest vars, so talking to Mapitoto after trying once will result in completion. 😉

File: scripts/globals/quests.lua (hunk 1 above)
Comment (TeoTwawki) [reply scripts/globals/quests.lua:485]: ~~he dat index stops at 158 and shows this quest as 157 so I dunno wtf we did to get these values...~~
_our packet is fubar and we're gonna wind up with migrations again later, awesome_

Comment (ibm2431) [reply scripts/globals/quests.lua:485]: More fuel for the dumpster fire that this one PR line has illuminated: every quest area is messed up at each interval of 32!

File: scripts/quests/full_speed_ahead.lua (hunk 2 above)
Comment (zach2good) [reply scripts/quests/full_speed_ahead.lua:50]: It has its own special effect which is a wrapper for regular mount and all the functions in `quests/full_speed_ahead.lua`. This handles the loss of the underlying mounted effect, it's own loss, ticks, power for difficulty etc.

I tested your scenario: nothing extra happens :+1:

**Review by ibm2431 — APPROVED** [review]
We'll have to fix our quest logs and packets later; I'm not going to dump that on you to fix before this can be merged!

**Review by cocosolos — APPROVED** [review]

**Pull request merged**